### ISSUE TITLE
Self-hosting feasibility

### ISSUE THREAD

**Issue by bo0tzz** [cachapa/tudo#25]
This app looks great, it's exactly what I've been looking for for a while! How feasible is it at the moment to run a tudo server on my own systems? The two limitations I see right now are:
1. Not being able to configure the server URL in the app
2. (non-critical) No docker images available

Is there an interest in making tudo easier to self-host? Are there any other current limitations I've missed? If it is wanted, I could probably contribute the necessary parts for building and releasing docker images for tudo_server (and perhaps for the web build of the client?)

Thank you!

**Comment by cachapa** [comment]
Currently the only way to self-host is by building a custom version of the app, and hardcoding the server uri in [config.uri](https://github.com/cachapa/tudo/blob/master/lib/config.dart).

At one point I wanted to make it configurable, but that requires a non-trivial amount of work since the app would need to start with a setup screen on first launch. It's not super complex though, it just takes some time.

The client should work fine on Flutter web. I tested it a while ago and the only issue is that the UI doesn't really scale well to desktop sizes. I made a few token efforts to implement a master/detail UX but ran out of time for that.

On the server side, a docker container would be ideal, I would be happy to accept contributions: https://github.com/cachapa/tudo_server/issues/1
